src/scripts/Controller.js

```diff
@@ -1200,20 +1201,71 @@ export default class SitemapController {
 		this._editSelector(selector, sitemap);
 	}
 
+	async showConfirmActionPanel(message) {
+		const $actionConfirm = ich.ActionConfirm();
+		$('#viewport').append($actionConfirm);
+		const $title = $('.modal-title');
+		const $subtitle = $('.modal-subtitle');
+		$title.attr('data-i18n', message.title_i18n);
+		$subtitle.attr('data-i18n', message.subtitle_i18n);
+		$actionConfirm.modal('show');
+		Translator.translatePage();
+
+		if (message.type === 'deleteSelector') {
+			$('#selector-id').append(message.title_replace['selector-id']);
+			if (message.subtitle_replace['child-count']) {
+				$('#child-count').append(message.subtitle_replace['child-count']);
+				$subtitle.show();
+			}
+		}
+		if (message.type === 'deleteSitemap') {
+			$('#sitemap-id').append(message.title_replace['sitemap-id']);
+		}
```

Comment:
Looks complicated.
We try to generalize confirmation panel but still have to switch on panel type to insert things.

You may combine `title_replace` and `subtitle_replace` into single object with span ids (or css selectors) as keys and texts to insert as values, then traverse it and perform insertions:
`Object.entries(replace).forEach((selector, text) => $(selector).append(text);`
I don't believe it simplifies code enough though.

Consider splitting `showConfirmActionPanel` function in two where one creates the panel and other handles interaction. Then in `deleteSelector` you can create panel, then insert id and child count, then await for interaction.